lib/__tests__/github.ts

```diff
@@ -48,6 +56,51 @@ describe('lib/github', () => {
     });
   });
 
+  describe('fetchGitHubRepoContentFromDownloadUrl()', () => {
+    const dest = '/path/to/destination';
+    const downloadUrl = 'https://github.com/repo/file';
+
+    it('should write text to the destination file', async () => {
+      const resp = {
+        headers: {
+          'content-type': 'text/plain',
+        },
+        data: Buffer.from('Sample text', 'utf-8'),
+        status: 200,
+        statusText: 'OK',
+        config: {},
+      };
+
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      fetchRepoFileByDownloadUrl.mockResolvedValue(resp);
+
+      await fetchGitHubRepoContentFromDownloadUrl(dest, downloadUrl);
+
+      expect(fs.outputFileSync).toHaveBeenCalledWith(dest, 'Sample text');
+    });
+
+    it('should write binary to the destination file', async () => {
+      const resp = {
+        headers: {
+          'content-type': 'application/octet-stream',
+        },
+        data: Buffer.from([0x01, 0x02, 0x03]),
+        status: 200,
+        statusText: 'OK',
+        config: {},
+      };
+
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```

Comment:
I'm having an issue with TypeScript here that I'd love some advice on:

TypeScript is basically requiring that we mock out everything in the `config` property, as so:

`config: {
  headers: {
    set: jest.fn(),
    etc. 
  }
}

Is that really necessary for this single test? It seems like overkill, so I disabled the error for now. Is there anything I'm missing?

Reply:
^^ Talked this over with Camden in Slack, and he agreed it's overkill to include all the data.